ext/pango/gstbasetextoverlay.c

```diff
@@ -2080,16 +2080,21 @@ gint x0, gint x1, gint y0, gint y1) \
 { \
   gint i, j;\
   guint8 *dest_ptr;\
+  gint alpha_offset[2] = { 3, 0 };\
   \
   dest_ptr = dest->data[0];\
   \
   for (i = y0; i < y1; i++) {\
     for (j = x0; j < x1; j++) {\
       gint y, y_pos, k;\
       y_pos = (i * 4 * overlay->width) + j * 4;\
-      for (k = OFFSET; k < 3+OFFSET; k++) {\
-        y = dest_ptr[y_pos + k] - overlay->shading_value;\
-        dest_ptr[y_pos + k] = CLAMP (y, 0, 255);\
+      if (dest_ptr[y_pos + alpha_offset[OFFSET]] == 0) {\
+        dest_ptr[y_pos + alpha_offset[OFFSET]] = overlay->shading_value;\
```

Comment:
Again, probably something that gets optimised, but consider putting overlay->shading_value into variable outside of the loop, just in case it isn't.

Reply:
the existing code uses `overlay->shading_value` directly as well, i just copied that